app/models/announcement.rb

```diff
@@ -23,7 +23,8 @@ def start_date_before_end_date
   end
   
   def self.current
-    where("end_date >= ?", Date.today).first
+    where("end_date >= :end_date AND start_date <= :start_date", {end_date: Date.today, start_date: Date.today}).order(:start_date).first
+    
```

Comment:
1) We are looking to see if there is a current announcement to display
2) In order of the announcement to be current, i.e. today's, the start_date must be today or before and the end_date must be today or after
3) When combined with non overlapping dates (an added condition in this PR) it leaves a unique message
4) the logic is also backward compatible, the original bug was due to an earlier announcement (lower id) blocking a true "current" announcement since sorting was by id when the earlier announcement was still valid in the future but not current

An enhancement would be to delete the cookie which hides the announcement

>> @@ -23,7 +23,8 @@ def start_date_before_end_date
>>   end
>> 
>>   def self.current
>> -    where("end_date >= ?", Date.today).first
>> +    where("end_date >= :end_date AND start_date <= :start_date",
>> {end_date: Date.today, start_date: Date.today}).order(:start_date).first
>> +
> 
> Can you explain this logic please?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/phran/forever-family-foundation/pull/51/files#r40028030